pkg/parse/run.go

```diff
@@ -120,6 +126,24 @@ func Run(ctx context.Context, p Parser) {
 			// state of backoff retry.
 			statusUpdateTimer.Reset(opts.statusUpdatePeriod) // Schedule status update attempt
 
+		// Execute the entire parse-apply-watch loop for a namespace event.
+		case <-nsEventTimer.C:
+			if nsController == nil {
```

Comment:
This is one of those situations where using reflect.Select might be a good idea. If the nsController isn't running/enabled then we don't need the nsEventTimer or this select case at all.

Reply:
If the nsConroller isn't running, we stop the timer.